gitprovider/client_options_test.go

```diff
@@ -0,0 +1,207 @@
+	o := &CommonClientOptions{}
+	for _, opt := range opts {
+		if err := opt.ApplyToCommonClientOptions(o); err != nil {
+			return nil, err
+		}
+	}
+	return o, nil
+}
+
+func withDomain(domain string) commonClientOption {
+	return &CommonClientOptions{Domain: &domain}
+}
+
+func withDestructiveAPICalls(destructiveActions bool) commonClientOption {
+	return &CommonClientOptions{EnableDestructiveAPICalls: &destructiveActions}
+}
+
+func withPreChainTransportHook(preRoundTripperFunc ChainableRoundTripperFunc) commonClientOption {
+	return &CommonClientOptions{PreChainTransportHook: preRoundTripperFunc}
+}
+
+func withPostChainTransportHook(postRoundTripperFunc ChainableRoundTripperFunc) commonClientOption {
+	return &CommonClientOptions{PostChainTransportHook: postRoundTripperFunc}
+}
+
+func dummyRoundTripper1(http.RoundTripper) http.RoundTripper { return nil }
+
+func Test_makeOptions(t *testing.T) {
+	tests := []struct {
+		name         string
+		opts         []commonClientOption
+		want         *CommonClientOptions
+		expectedErrs []error
+	}{
+		{
+			name: "no options",
+			want: &CommonClientOptions{},
+		},
+		{
+			name: "withDomain",
+			opts: []commonClientOption{withDomain("foo")},
+			want: &CommonClientOptions{Domain: StringVar("foo")},
+		},
+		{
+			name:         "withDomain, empty",
+			opts:         []commonClientOption{withDomain("")},
+			expectedErrs: []error{ErrInvalidClientOptions},
+		},
+		{
+			name:         "withDomain, duplicate",
+			opts:         []commonClientOption{withDomain("foo"), withDomain("bar")},
+			expectedErrs: []error{ErrInvalidClientOptions},
+		},
+		{
+			name: "withDestructiveAPICalls",
+			opts: []commonClientOption{withDestructiveAPICalls(true)},
+			want: &CommonClientOptions{EnableDestructiveAPICalls: BoolVar(true)},
+		},
+		{
+			name:         "withDestructiveAPICalls, duplicate",
+			opts:         []commonClientOption{withDestructiveAPICalls(true), withDestructiveAPICalls(false)},
+			expectedErrs: []error{ErrInvalidClientOptions},
+		},
+		{
+			name: "withPreChainTransportHook",
+			opts: []commonClientOption{withPreChainTransportHook(dummyRoundTripper1)},
+			want: &CommonClientOptions{PreChainTransportHook: dummyRoundTripper1},
+		},
+		{
+			name:         "withPreChainTransportHook, duplicate",
+			opts:         []commonClientOption{withPreChainTransportHook(dummyRoundTripper1), withPreChainTransportHook(dummyRoundTripper1)},
+			expectedErrs: []error{ErrInvalidClientOptions},
+		},
+		{
+			name: "withPostChainTransportHook",
+			opts: []commonClientOption{withPostChainTransportHook(dummyRoundTripper1)},
+			want: &CommonClientOptions{PostChainTransportHook: dummyRoundTripper1},
+		},
+		{
+			name:         "withPostChainTransportHook, duplicate",
+			opts:         []commonClientOption{withPostChainTransportHook(dummyRoundTripper1), withPostChainTransportHook(dummyRoundTripper1)},
+			expectedErrs: []error{ErrInvalidClientOptions},
+		},
+		/*{
+			name: "WithDestructiveAPICalls",
```

Comment:
Is there a reason why these are commented?

Reply:
oops, dead code, will remove